src/Microsoft.PowerShell.ConsoleHost/host/msh/ProgressNode.cs

```diff
@@ -387,6 +387,12 @@ internal static bool IsMinimalProgressRenderingEnabled()
                 maxWidth = PSStyle.Instance.Progress.MaxWidth;
             }
 
+            // if the activity is really long, only use up to half the width
+            if (Activity.Length > maxWidth / 2)
+            {
+                Activity = Activity.Substring(0, maxWidth / 2) + PSObjectHelper.Ellipsis;
+            }
+
             // 4 is for the extra space and square brackets below and one extra space
             int barWidth = maxWidth - Activity.Length - indentation - 4;
```

Comment:
1. It doesn't cover all edge cases again.
2. We change Activity - what if user will expand Window width?
3. What about adding "..." on Activity end?

Reply:
The `maxWidth` comes from [_bufsize](https://github.com/PowerShell/PowerShell/blob/658837323599ab1c7a81fe66fcd43f7420e4402b/src/Microsoft.PowerShell.ConsoleHost/host/msh/ProgressPane.cs#L218) which gets [retrieved](https://github.com/PowerShell/PowerShell/blob/658837323599ab1c7a81fe66fcd43f7420e4402b/src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostRawUserInterface.cs#L348) from the console.  So if the Window width changes, this value will change and is read every time progress is rendered.

Adding `...` to end of Activity is not an issue as the max length of Activity is now half the window width + 1 character which in most cases will be fine.  There is an edge case where the window width is extremely narrow which I can fix.